x/migration/keeper/msg_server_claim_morse_application.go

```diff
@@ -137,12 +138,68 @@ func (k msgServer) ClaimMorseApplication(ctx context.Context, msg *migrationtype
 		morseClaimableAccount,
 	)
 
+	claimedAppStake := morseClaimableAccount.GetApplicationStake()
+	sharedParams := k.sharedKeeper.GetParams(ctx)
+	sessionEndHeight := sharedtypes.GetSessionEndHeight(&sharedParams, sdkCtx.BlockHeight())
+	claimedUnstakedBalance := morseClaimableAccount.GetUnstakedBalance()
+
+	currentSessionStart := sharedtypes.GetSessionStartHeight(&sharedParams, sdkCtx.BlockHeight())
+	previousSessionEnd := sharedtypes.GetSessionEndHeight(&sharedParams, currentSessionStart-1)
+
 	// Query for any existing application stake prior to staking.
 	preClaimAppStake := cosmostypes.NewInt64Coin(volatile.DenomuPOKT, 0)
 	foundApp, isFound := k.appKeeper.GetApplication(ctx, shannonAccAddr.String())
 	if isFound {
 		preClaimAppStake = *foundApp.Stake
 	}
+	postClaimAppStake := preClaimAppStake.Add(morseClaimableAccount.GetApplicationStake())
+
+	// TODO_IN_THIS_COMMIT: comment...
+	// ... application began unbonding on Morse and the unbonding period has since elapsed...
+	// ... both stake and unstatked balance were minted to the shannonDestAddr account...
+	// ... no application was staked...
+	// - use a lookup table to get the est. block times per network...
+	// - use the shared params and the block time to calculate the unstake session end height...
+	// - emit an unbonding start event...
+	if !morseClaimableAccount.UnstakingCompletionTime.IsZero() {
```

Comment:
Shouldn't this be less than or equal to zero (given the -1 return value in the function)?

Reply:
I factored out some methods so that this is easier to read:

```go
	if morseClaimableAccount.IsUnbonding() {
		if morseClaimableAccount.SecondsUntilUnbonded() <= 0 {
```